examples/triage/baseaddress.cpp

```diff
@@ -0,0 +1,427 @@
+	if (!mappedView)
+		return;
+
+	auto fileMetadata = m_view->GetFile();
+	if (!fileMetadata)
+		return;
+
+	uint64_t address;
+	string errorStr;
+	if (!BinaryNinja::BinaryView::ParseExpression(m_view, m_reloadBase->text().toStdString(), address, 0, errorStr))
+	{
+		m_status->setText(QString("Invalid rebase address (%1)").arg(QString::fromStdString(errorStr)));
+		return;
+	}
+
+	if (!fileMetadata->Rebase(mappedView, address))
+		return;
+
+	BinaryNinja::Settings::Instance()->Set("analysis.mode", "full", mappedView);
+	mappedView->Reanalyze();
+
+	auto frame = ViewFrame::viewFrameForWidget(this);
+	if (!frame)
+		return;
+
+	auto fileContext = frame->getFileContext();
+	if (!fileContext)
+		return;
+
+	auto uiContext = UIContext::contextForWidget(this);
+	if (!uiContext)
+		return;
+
+	uiContext->recreateViewFrames(fileContext);
+	fileContext->refreshDataViewCache();
+
+	auto newFrame = ViewFrame::viewFrameForWidget(this);
+	if (!newFrame)
+		return;
+
+	auto view = newFrame->getCurrentViewInterface();
+	if (!view)
+		return;
+
+	auto data = view->getData();
+	if (!data)
+		return;
+
+	if (!view->navigate(address))
+		data->Navigate("Linear:Mapped", address);
+}
+
+
+void BaseAddressDetectionWidget::CreateAdvancedSettingsGroup()
+{
+	int32_t row = 0;
+	int32_t column = 0;
+	auto grid = new QGridLayout();
+
+	grid->addWidget(new QLabel("Min. String Length:"), row, column, Qt::AlignLeft);
+	m_inputs.StrlenLineEdit = new QLineEdit("0n10");
+	grid->addWidget(m_inputs.StrlenLineEdit, row, column + 1, Qt::AlignLeft);
+
+	grid->addWidget(new QLabel("Alignment:"), row, column + 2, Qt::AlignLeft);
+	m_inputs.AlignmentLineEdit = new QLineEdit("0n1024");
+	grid->addWidget(m_inputs.AlignmentLineEdit, row++, column + 3, Qt::AlignLeft);
+
+	grid->addWidget(new QLabel("Lower Boundary:"), row, column, Qt::AlignLeft);
+	m_inputs.LowerBoundary = new QLineEdit("0x0");
+	grid->addWidget(m_inputs.LowerBoundary, row, column + 1, Qt::AlignLeft);
+
+	grid->addWidget(new QLabel("Upper Boundary:"), row, column + 2, Qt::AlignLeft);
+	m_inputs.UpperBoundary = new QLineEdit("0xffffffffffffffff");
+	grid->addWidget(m_inputs.UpperBoundary, row++, column + 3, Qt::AlignLeft);
+
+	grid->addWidget(new QLabel("Points Of Interest:"), row, column, Qt::AlignLeft);
+	auto poiList = QStringList() << "All" << "Strings only" << "Functions only";
+	m_inputs.POIBox = new QComboBox(this);
+	m_inputs.POIBox->addItems(poiList);
+	grid->addWidget(m_inputs.POIBox, row, column + 1, Qt::AlignLeft);
+
+	grid->addWidget(new QLabel("Max Pointers:"), row, column + 2, Qt::AlignLeft);
+	m_inputs.MaxPointersPerCluster = new QLineEdit("0n128");
+	grid->addWidget(m_inputs.MaxPointersPerCluster, row++, column + 3, Qt::AlignLeft);
+
+	m_advancedSettingsGroup = new ExpandableGroup(grid);
+	m_advancedSettingsGroup->setTitle("Advanced Settings");
+}
+
+
+BaseAddressDetectionWidget::BaseAddressDetectionWidget(QWidget* parent, BinaryNinja::Ref<BinaryNinja::BinaryView> bv)
```

Comment:
You should be initializing the parent `: QWidget(parent)` otherwise none of the objects stored in here know when to get cleaned up